src/entities/factories/Factory.ts

```diff
@@ -2,30 +2,54 @@ import { merge } from 'lodash';
 import { Entity } from '../Entity';
 import { Context } from '../../Context';
 
+/**
+ * Represents an entity
+ */
 export interface EntityClass<T, U> {
   new (params: T & U, context: Context): Entity<T>;
+
+  /**
+   * unserialize serialized entity information
+   */
   unserialize(uid: string): U;
+
+  /**
+   * generate a unique identifier for an entity
+   */
   generateId(identifiers: U): string;
 }
 
+/**
+ * Abstract representing a factory that will build up properties of an entity
+ */
 export abstract class Factory<EntityType extends Entity<T>, T extends any, U extends any> {
   public cache: {
     [key: string]: EntityType | undefined;
   } = {};
 
   public context: Context;
 
+  /**
+   * Defined entity class
```

Comment:
Lower case

Reply:
Also, change it to

Entity class that this Factory is in charge of generating and caching